### PR TITLE
feat(frontend): settings page with users and key management

### PR DESCRIPTION
**Pull request by shaleynikov**
WIP Settings page & server

### CODE REVIEW

File: webapp/javascript/services/users.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const response = await request('/api/users');
+
+  try {
+    if (response.isOk) {
+      console.log('Parsing', response.value);
+      return parse(response.value);
+    }
+  } catch (e) {
+    console.error(e);
+  }
```
Comment (eh-am) [inline webapp/javascript/services/users.ts:23]: Sorry, why was this needed? The whole point of using `Result` is to not deal with exceptions, so why was this throwing an exception? Let me know, maybe we messed something up in the base implementation.

File: webapp/javascript/redux/reducers/settings.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  // There's no 'loading'
+  users:
+    | { type: 'loaded'; data: Users }
+    | { type: 'reloading'; data: Users }
+    | { type: 'failed'; data: Users };
+}
+
+// Define the initial state using that type
+const initialState: SettingsRootState = {
+  users: { type: 'loaded', data: (window as any).initialState.users },
```
Comment (eh-am) [inline webapp/javascript/redux/reducers/settings.ts:18]: Wait is this populated from the server?

File: webapp/javascript/components/Settings/Users/UserAddForm.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    };
+    console.log(JSON.stringify(data));
+    fetch('/api/users', {
+      method: 'POST',
+      body: JSON.stringify(data),
+      headers: [
+        ['Content-Type', 'application/json'],
+        ['Accept', 'application/json'],
+      ],
+    })
```
Comment (eh-am) [inline webapp/javascript/components/Settings/Users/UserAddForm.tsx:39]: I assume this is due to the PR being a WIP, but I think this should be moved to the `services/users`?

File: webapp/javascript/components/Settings/Users/UserAddForm.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      .then((r) => r.json())
+      .then((r) => {
+        console.log(r);
+        return r;
+      })
+      .then(() => {
+        setForm({});
+        dispatch(reloadUsers());
```
Comment (eh-am) [inline webapp/javascript/components/Settings/Users/UserAddForm.tsx:47]: And all this call to a thunk

File: webapp/javascript/components/Settings/Users/index.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+function Users() {
+  const dispatch = useAppDispatch();
+  const users = useAppSelector(selectUsers);
+  console.log('Rendering users');
+  useEffect(() => {
+    if (typeof users === 'undefined') {
+      console.log('Reloading users');
+      dispatch(reloadUsers());
+    }
+  });
```
Comment (eh-am) [inline webapp/javascript/components/Settings/Users/index.tsx:35]: Is this the case where we load users once?
If so i think it can be written as 
```suggestion
  useEffect(() => {
      dispatch(reloadUsers());
  }, []);
```

File: webapp/javascript/components/Settings/index.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      <h1>Settings</h1>
+      <nav>
+        <ul className={styles.settingsNav}>
+          <li>
+            <NavLink
+              to={`${url}`}
+              exact
+              className={(isActive) =>
+                styles.navLink + (isActive ? ` ${styles.navLinkActive}` : '')
+              }
```
Comment (eh-am) [inline webapp/javascript/components/Settings/index.tsx:30]: Can we simplify these with that https://www.npmjs.com/package/classnames lib?

File: webapp/javascript/components/Sidebar.tsx [hunk 7]
```diff
@@ -71,6 +72,10 @@ export default function Sidebar2() {
     isRouteActive('/adhoc-single') ||
     isRouteActive('/adhoc-comparison') ||
     isRouteActive('/adhoc-comparison-diff');
+  const isSettingsActive =
+    isRouteActive('/settings') ||
+    isRouteActive('/settings/users') ||
+    isRouteActive('/settings/api-keys');
```
Comment (eh-am) [inline webapp/javascript/components/Sidebar.tsx:78]: Can we be lazy and use `isRouteActive(/settings*/)` that requires updating that `isRouteActive` function to support regex but it should be fine?

My worry is having to come back to update this function every time we add a new tab to the settings.

**Comment by eh-am** [comment]
Also would be nice to use https://github.com/pyroscope-io/pyroscope/blob/main/stories/Notifications.stories.tsx when the requests fail!

File: webapp/javascript/components/Settings/Users/UserAddForm.tsx (hunk 3 above)
Comment (shaleynikov) [reply webapp/javascript/components/Settings/Users/UserAddForm.tsx:39]: Yeah, ofc all api related code goes to `services`. I just was at high speed :-)

File: webapp/javascript/redux/reducers/settings.ts (hunk 2 above)
Comment (shaleynikov) [reply webapp/javascript/redux/reducers/settings.ts:18]: Actually It looks like I left something uncommitted

File: webapp/javascript/services/users.ts (hunk 1 above)
Comment (shaleynikov) [reply webapp/javascript/services/users.ts:23]: it may throw an exception once it doesn't pass `zod` validation inside `parse(response.value);`

**Comment by kolesnikovae** [comment]
LGTM. Thank you @shaleynikov, nice job! I'll take care of the backend stuff.

File: webapp/templates/login.html [hunk 8]
```diff
@@ -13,14 +13,66 @@
...
+          <div>
+            <div><label>Email or username</label></div>
+            <input type="text" name="user" autocapitalize="none" placeholder="email or username" id="username" />
+          </div>
+          <div>
+            <div><label>Password</label></div>
+            <input type="password" name="password" aria-label="password input field" id="password" autocomplete="current-password"  placeholder="password" />
+          </div>
+          <div>
+            <button type="button" class="sign-in-button" onClick="login()">Log In</button>
```
Comment (petethepig) [inline webapp/templates/login.html:60]: I'd move `login` function to `form onSubmit`, and make the button `type=submit`. that way it works when people press enter and things like that.

Comment (shaleynikov) [reply webapp/templates/login.html:60]: Sure, I'd change all other forms same way.
